extensions/aws/processors/PutS3Object.h

```diff
@@ -0,0 +1,188 @@
+  constexpr const char *CA_CENTRAL_1 = "ca-central-1";
+
+}  // namespace region
+
+class PutS3Object : public core::Processor {
+ public:
+  static constexpr char const* ProcessorName = "PutS3Object";
+
+  static const std::set<std::string> CANNED_ACLS;
+  static const std::set<std::string> REGIONS;
+  static const std::set<std::string> STORAGE_CLASSES;
+  static const std::set<std::string> SERVER_SIDE_ENCRYPTIONS;
+
+  // Supported Properties
+  static const core::Property ObjectKey;
+  static const core::Property Bucket;
+  static const core::Property ContentType;
+  static const core::Property AccessKey;
+  static const core::Property SecretKey;
+  static const core::Property CredentialsFile;
+  static const core::Property AWSCredentialsProviderService;
+  static const core::Property StorageClass;
+  static const core::Property ServerSideEncryption;
+  static const core::Property Region;
+  static const core::Property CommunicationsTimeout;
+  static const core::Property FullControlUserList;
+  static const core::Property ReadPermissionUserList;
+  static const core::Property ReadACLUserList;
+  static const core::Property WriteACLUserList;
+  static const core::Property CannedACL;
+  static const core::Property EndpointOverrideURL;
+  static const core::Property ProxyHost;
+  static const core::Property ProxyPort;
+  static const core::Property ProxyUsername;
+  static const core::Property ProxyPassword;
+
+  // Supported Relationships
+  static const core::Relationship Failure;
+  static const core::Relationship Success;
+
+  explicit PutS3Object(std::string name, minifi::utils::Identifier uuid = minifi::utils::Identifier())
+      : PutS3Object(name, uuid, minifi::utils::make_unique<aws::s3::S3Wrapper>()) {
+  }
+
+  explicit PutS3Object(std::string name, minifi::utils::Identifier uuid, std::unique_ptr<aws::s3::S3WrapperBase> s3_wrapper)
+      : core::Processor(std::move(name), uuid)
+      , s3_wrapper_(std::move(s3_wrapper)) {
+  }
+
+  ~PutS3Object() override = default;
+
+  bool supportsDynamicProperties() override { return true; }
+  void initialize() override;
+  void onSchedule(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSessionFactory> &sessionFactory) override;
+  void onTrigger(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSession> &session) override;
+
+  class ReadCallback : public InputStreamCallback {
+   public:
+    static const uint64_t MAX_SIZE = 5UL * 1024UL * 1024UL * 1024UL;  // 5GB limit on AWS
+    static const uint64_t BUFFER_SIZE = 4096;
+
+    ReadCallback(uint64_t flow_size, const minifi::aws::s3::PutObjectRequestParameters& options, aws::s3::S3WrapperBase* s3_wrapper)
+      : flow_size_(flow_size)
+      , options_(std::move(options))
+      , s3_wrapper_(s3_wrapper) {
+    }
+
+    ~ReadCallback() = default;
+
+    int64_t process(std::shared_ptr<io::BaseStream> stream) {
```

Comment:
we should add `override` to `~ReadCallback` and `process`

Reply:
As there is no need for override in the destructor I removed it and added the override to the process in [2e71a70](https://github.com/apache/nifi-minifi-cpp/pull/915/commits/2e71a707a31f5564360c40c7bfdd38c3d97302ef)